### ISSUE TITLE
Documentation on new default filters is inconsistent with implementation

### ISSUE THREAD

**Issue by carstenlenz** [playframework/playframework#7823]
Hello, 

The documentation on the new `play.filters.enabled` configuration says:
> In Play itself, play.filters.enabled is an empty list. However, the filters library is automatically loaded in SBT as an AutoPlugin called PlayFilters, and will **append** the following values to the play.filters.enabled property

The implementation of this does not **append** as it is says but sets it:
```
play.filters {
  # Default list of enabled filters, configured by play.api.http.EnabledFilters
  enabled = [
    play.filters.csrf.CSRFFilter,
    play.filters.headers.SecurityHeadersFilter,
    play.filters.hosts.AllowedHostsFilter
  ]
 ...
}
```
What I tried is to add a new filter in my libraries `reference.conf` by appending like:
```
play.filters.enabled += Myfilter
```
But depending on classpath ordering my append gets overwritten or not. 
To get it to work I have to set it explicitly in the `application.conf` of the app that depends on the library.

So the question is essentially: Should it be allowed for libraries to add default filters like they can add modules (`play.modules.enabled +=`) ? If not the docs should be more clear about this - if they should be allowed then the current behaviour looks like an issue to me. 

Thank you for you awesome work :)
Cheers
Carsten

**Comment by mkurz** [comment]
Sounds reasonable for me. Opened a pull request: #7824

**Comment by carstenlenz** [comment]
Thanks :)

**Comment by mkurz** [comment]
@carstenlenz This will be fixed in the upcoming Play 2.6.4 release, which should happen soon (I think). Let us know if it does work for you then. Thanks for reporting!

**Comment by mkurz** [comment]
@carstenlenz Play 2.6.5 is out with the fix (don't use 2.6.4)

**Comment by carstenlenz** [comment]
I just upgraded our apps.
Works for me! 👍 

Thank you very much :)
